community/kernel/src/main/java/org/neo4j/kernel/impl/core/NodeProxy.java

```diff
@@ -779,6 +789,25 @@ public int getDegree( RelationshipType type, Direction direction )
         }
     }
 
+    //Part of NodeValue API
+
+    @Override
+    public TextArray labels()
+    {
+        ArrayList<String> ls = new ArrayList<>();
+        for ( Label label : getLabels() )
+        {
+            ls.add( label.name() );
+        }
+        return Values.stringArray( ls.toArray( new String[ls.size()] ) );
```

Comment:
This seems like the wrong way around - if the implementation is the way it is here, with `NodeProxy` being both a `NodeValue` and a `Node`, surely the `getLabels()` method should be based on this method rather than the other way around.